noir-projects/noir-contracts/contracts/app/nft_contract/src/main.nr

```diff
@@ -208,19 +208,10 @@ pub contract NFT {
             sender_and_completer,
         );
 
-        // We can't simply return the partial note because we won't be able to later on verify that it was created
-        // correctly (e.g. that the storage slot corresponds to the owner, and that we're using the `private_nfts`
-        // state variable and not a different one) once this information is hidden in the partial note commitment.
-        // We also want to verify that only the caller of this function can complete the partial note.
-        //
-        // We achieve this by storing a `validity_commitment` in the nullifier tree. This commitment contains all
-        // the necessary information to verify the completer and the partial note. Later, if we successfully
-        // read this commitment from the nullifier tree, we know that the partial note is valid.
-        let validity_commitment = partial_note.compute_validity_commitment(sender_and_completer);
-
-        // The validity commitment can be safely pushed to the nullifier tree since it uses its own separator,
-        // making collisions with actual note nullifiers practically impossible.
-        context.push_nullifier(validity_commitment);
+        // We insert the partial note validity commitment into the nullifier tree to be later on able to verify that
+        // the partial note and completer are legitimate. See function docs of `insert_validity_commitment` for more
+        // details.
+        partial_note.insert_validity_commitment(context, sender_and_completer);
```

Comment:
Do you think it'd be too much if we made this part of the partial note creation? It'd put the entire process inside the partial note module - from the outside you just create and complete.

Reply:
We want to force the dev to insert it and doing so manually in the contract here relies on him not forgetting it. So I think moving it the the note creation makes sense. Addressed it in https://github.com/AztecProtocol/aztec-packages/pull/14432/commits/424f69d3f59ec2b111fd1cffb46dcc5d9bbf4745

I just renamed the function from `partial` to `setup_partial_note` as `partial` to me sounds less like a piece of complex involved thing (that the `partial` function has become) than `setup_partial_note`.